packages/http-client-csharp/emitter/src/emitter.ts

```diff
@@ -150,25 +152,40 @@ export async function $onEmit(context: EmitContext<NetEmitterOptions>) {
         const command = `dotnet --roll-forward Major ${generatorPath} ${outputFolder} -p ${options["plugin-name"]}${constructCommandArg(newProjectOption)}${constructCommandArg(existingProjectOption)}${constructCommandArg(debugFlag)}`;
         Logger.getInstance().info(command);
 
-        const result = await execAsync(
-          "dotnet",
-          [
-            "--roll-forward",
-            "Major",
-            generatorPath,
-            outputFolder,
-            "-p",
-            options["plugin-name"],
-            newProjectOption,
-            existingProjectOption,
-            debugFlag,
-          ],
-          { stdio: "inherit" },
-        );
-        if (result.exitCode !== 0) {
-          if (result.stderr) Logger.getInstance().error(result.stderr);
-          if (result.stdout) Logger.getInstance().verbose(result.stdout);
-          throw new Error(`Failed to generate SDK. Exit code: ${result.exitCode}`);
+        try {
+          const result = await execAsync(
+            "dotnet",
+            [
+              "--roll-forward",
+              "Major",
+              generatorPath,
+              outputFolder,
+              "-p",
+              options["plugin-name"],
+              newProjectOption,
+              existingProjectOption,
+              debugFlag,
+            ],
+            { stdio: "inherit" },
+          );
+          if (result.exitCode !== 0) {
+            if (result.stderr) Logger.getInstance().error(result.stderr);
+            if (result.stdout) Logger.getInstance().verbose(result.stdout);
+            throw new Error(`Failed to generate SDK. Exit code: ${result.exitCode}`);
+          }
+        } catch (error: any) {
+          if (error && "code" in (error as {}) && error["code"] === "ENOENT") {
+            reportDiagnostic(sdkContext.program, {
+              code: "dependency-runtime-missing",
```

Comment:
i think `runtime-dependency-missing` makes more sense.
Has this code been reviewed?

Reply:
update to `runtime-dependency-missing`. The code is for csharp-emitter only. if needed, we can review by .net arch.